presto-main/src/main/java/com/facebook/presto/dispatcher/DispatchManager.java

```diff
@@ -239,6 +246,35 @@ private <C> void createQueryInternal(QueryId queryId, String slug, int retryCoun
         }
     }
 
+    private Session checkPermissions(Session session)
```

Comment:
I don't think we want to control this by session property.  Since that would allow users to control their permission checks.  I think it should only be controllable by config property.

Reply:
This is a good point. I was thinking that it may be fine to add this session property at the start so that we can easily test by enabling and disabling this in case of issues. Later once we are confident of this working, we should definitely remove this session property. 
I am okay with not having this session property to start with as well.

Reply:
I'd prefer leaving it out from the start.

Reply:
Then it will be simpler to use config to control the flag. Will update the logic.